### PR TITLE
Zohar fix reward zero

### PR DESCRIPTION
**Pull request by Zohar97**

### CODE REVIEW

**Review by manhlx3006 — DISMISSED** [review]

File: contracts/liquidityMining/KyberRewardLockerV2.sol [hunk 1]
```diff
@@ -118,41 +118,59 @@ contract KyberRewardLockerV2 is IKyberRewardLockerV2, PermissionAdmin {
...
+        token.safeTransferFrom(msg.sender, account, quantity);
       }
+      // append new schedule
+      schedules.data[schedulesLength] = VestingSchedule({
+        startTime: startTime.toUint64(),
+        endTime: endTime.toUint64(),
+        quantity: quantity.toUint128(),
+        vestedQuantity: quantity.toUint128()
+      });
+      accountVestedBalance[account][token] = accountVestedBalance[account][token].add(
```
Comment (manhlx3006) [inline contracts/liquidityMining/KyberRewardLockerV2.sol:141]: Should still emit all related events for add new schedule + claim rewards

Comment (Zohar97) [reply contracts/liquidityMining/KyberRewardLockerV2.sol:141]: fixed

File: contracts/liquidityMining/KyberRewardLockerV2.sol [hunk 2]
```diff
@@ -118,41 +118,60 @@ contract KyberRewardLockerV2 is IKyberRewardLockerV2, PermissionAdmin {
...
+          lastSchedule.quantity = uint256(lastSchedule.quantity).add(quantity).toUint128();
+          accountEscrowedBalance[account][token] = accountEscrowedBalance[account][token].add(
+            quantity
+          );
+          emit VestingEntryQueued(schedulesLength - 1, token, account, quantity);
+          return;
+        }
+      }
+       // append new schedule
+      schedules.data[schedulesLength] = VestingSchedule({
```
Comment (thongtrungtran) [inline contracts/liquidityMining/KyberRewardLockerV2.sol:165]: same with above

**Review by thongtrungtran — APPROVED** [review]

**Review by thongtrungtran — DISMISSED** [review]

File: contracts/liquidityMining/KyberRewardLockerV2.sol [hunk 3]
```diff
@@ -118,41 +118,60 @@ contract KyberRewardLockerV2 is IKyberRewardLockerV2, PermissionAdmin {
...
-      startTime: startTime.toUint64(),
-      endTime: endTime.toUint64(),
-      quantity: quantity.toUint128(),
-      vestedQuantity: 0
-    });
     schedules.length = schedulesLength + 1;
-    // record total vesting balance of user
-    accountEscrowedBalance[account][token] = accountEscrowedBalance[account][token].add(quantity);
-
     emit VestingEntryCreated(token, account, startTime, endTime, quantity, schedulesLength);
```
Comment (manhlx3006) [inline contracts/liquidityMining/KyberRewardLockerV2.sol:175]: Should remain the correct order of the event, VestingEntryQueued before Vested

**Review by thongtrungtran — APPROVED** [review]

File: contracts/liquidityMining/KyberRewardLockerV2.sol [hunk 4]
```diff
@@ -118,42 +118,62 @@ contract KyberRewardLockerV2 is IKyberRewardLockerV2, PermissionAdmin {
...
-        return;
+    if (vestingDuration == 0) {
+      // append new schedule
+      schedules.data[schedulesLength] = VestingSchedule({
+        startTime: startTime.toUint64(),
+        endTime: endTime.toUint64(),
+        quantity: quantity.toUint128(),
+        vestedQuantity: quantity.toUint128()
+      });
+      accountVestedBalance[account][token] = accountVestedBalance[account][token].add(
```
Comment (manhlx3006) [inline contracts/liquidityMining/KyberRewardLockerV2.sol:133]: where is the update of accountEscrowedBalance[account][token] ?

**Review by manhlx3006 — CHANGES_REQUESTED** [review]

**Review by manhlx3006 — APPROVED** [review]

**Pull request merged**